wgsl/index.bs

```diff
@@ -132,13 +132,14 @@ TODO: What indicates the end of a line?  (E.g. A line ends at the next linefeed
   <thead>
     <tr><th>Token<th>Definition
   </thead>
-  <tr><td>`FLOAT_LITERAL`<td>`(-?[0-9]*.[0-9]+ | -?[0-9]+.[0-9]*)((e|E)(+|-)?[0-9]+)?`
+  <tr><td>`DECIMAL_FLOAT_LITERAL`<td>`(-?[0-9]*.[0-9]+ | -?[0-9]+.[0-9]*)((e|E)(+|-)?[0-9]+)?`
+  <tr><td>`HEX_FLOAT_LITERAL`<td>`-?0x([0-9a-fA-F]*.?[0-9a-fA-F]+ | [0-9a-fA-F]+.[0-9a-fA-F]*)(p|P)(+|-)?[0-9]+`
```

Comment:
Comparing to C99 draft (http://www.open-std.org/jtc1/sc22/WG14/www/docs/n1256.pdf 6.4.42):

- should allow both 0x and 0X.  This worth raising to the group in a new issue.  I filed #1453
- the significand expression can be simplified to:
      
        ([0-9a-fA-F]*.[0-9a-fA-F]+ | [0-9a-fA-F]+.)

In fact, the same simplification can be applied to the decimal float literal.

- C99 doesn't include the leading minus.

The last two points are already tracked in https://github.com/gpuweb/gpuweb/issues/775